diff --git a/src/edit.c b/src/edit.c
--- a/src/edit.c
+++ b/src/edit.c
@@ -8101,6 +8101,8 @@
 	if (im_on)
 	    im_set_active(TRUE);
 # endif
+	if (regname == '=')
+	    u_sync(TRUE); /* sync undo, so the effect of setline() can be undone! */
     }
     if (regname == NUL || !valid_yank_reg(regname, FALSE))
     {
diff --git a/src/testdir/test61.in b/src/testdir/test61.in
--- a/src/testdir/test61.in
+++ b/src/testdir/test61.in
@@ -84,6 +84,16 @@
 ggO---:0put b
 ggO---:0put a
 ggO---:w >>test.out
+:so small.vim
+:set nocp
+:enew!
+oa
+:set ul=100
+ob
+:set ul=100
+o1a2=setline('.','1234')
+
+uu:%w >>test.out
 :qa!
 ENDTEST
 
diff --git a/src/testdir/test61.ok b/src/testdir/test61.ok
--- a/src/testdir/test61.ok
+++ b/src/testdir/test61.ok
@@ -41,3 +41,6 @@
 two
 two
 three
+
+a
+b
