--- configure.ac.orig	2013-08-05 20:11:27.000000000 -0400
+++ configure.ac	2013-08-05 20:17:32.000000000 -0400
@@ -484,7 +484,7 @@
                 PYTHON_CFLAGS="-I$PYTHON_INCLUDE"
                 AC_MSG_RESULT(found)
                 PYTHON_LIB=`$PYTHON -c "import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_config_var('LIBPL'))"`
-                PYTHON_LFLAGS="-lpython$PYTHON_VERSION "`$PYTHON -c "import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_config_var('LIBS')+' '+distutils.sysconfig.get_config_var('SYSLIBS')+' '+distutils.sysconfig.get_config_var('LINKFORSHARED'))"`
+                PYTHON_LFLAGS="`$PYTHON-config --ldflags`"
                 AC_MSG_CHECKING(for Python library)
                 if test -r "$PYTHON_LIB/libpython$PYTHON_VERSION.so"; then
                    PYTHON_LFLAGS="-L$PYTHON_LIB $PYTHON_LFLAGS"
--- cmake/FindPython.cmake.orig	2013-08-05 20:24:53.000000000 -0400
+++ cmake/FindPython.cmake	2013-08-05 20:25:26.000000000 -0400
@@ -57,7 +57,7 @@
     )
 
   EXECUTE_PROCESS(
-    COMMAND ${PYTHON_EXECUTABLE} -c "import sys; from distutils.sysconfig import *; sys.stdout.write(get_config_var('LINKFORSHARED'))"
+    COMMAND ${PYTHON_EXECUTABLE}-config --ldflags
     OUTPUT_VARIABLE PYTHON_LFLAGS
     )
 
