Description: fix for typos in manpages.
Author: Nicolas Boulenguez <nicolas.boulenguez@free.fr>

Index: b/fortune/fortune-man.part1
===================================================================
--- a/fortune/fortune-man.part1	2012-07-07 02:27:47.000000000 +0200
+++ b/fortune/fortune-man.part1	2012-07-07 02:29:52.000000000 +0200
@@ -58,7 +58,7 @@
 When
 .B fortune
 is run with no arguments it prints out a random epigram. Epigrams are
-divided into several categories, where each category is sub-divided
+divided into several categories, where each category is sub\-divided
 into those which are potentially offensive and those which are not.
 .SS Options
 The options are as follows:
@@ -104,7 +104,7 @@
 .BR fortunes , 
 i.e. the filenames themselves, in parentheses.  This can be useful if you
 wish to remove the gathered matches from their original files, since each
-filename-record will precede the records from the file it names.
+filename\-record will precede the records from the file it names.
 .RE
 .TP
 .BI "\-n " length
@@ -112,7 +112,7 @@
 ``short'' (the default is 160).  All fortunes longer than this are
 considered ``long''.  Be careful!  If you set the length too short and
 ask for short fortunes, or too long and ask for long ones, fortune goes
-into a never-ending thrash loop.
+into a never\-ending thrash loop.
 .TP
 .B \-o
 Choose only from potentially offensive aphorisms.  The \-o option is
@@ -130,10 +130,10 @@
 .RS
 \&... let us keep in mind the basic governing philosophy of The
 Brotherhood, as handsomely summarized in these words: we believe in
-healthy, hearty laughter -- at the expense of the whole human race, if
+healthy, hearty laughter \-\- at the expense of the whole human race, if
 needs be.  Needs be.
 .RS
---H. Allen Smith, "Rude Jokes"
+\-\-H. Allen Smith, "Rude Jokes"
 .RE
 .RE
 .TP
@@ -154,7 +154,7 @@
 read before the screen is cleared.
 .TP
 .B \-u
-Don't translate UTF-8 fortunes to the locale when searching or translating.
+Don't translate UTF\-8 fortunes to the locale when searching or translating.
 .PP
 The user may specify alternate sayings.  You can specify a specific
 file, a directory which contains one or more files, or the special word 
@@ -173,26 +173,26 @@
 selecting from one of them will be based on their relative sizes.
 .PP
 As an example, given two databases 
-.IR funny " and " not-funny ", with " funny
+.IR funny " and " not\-funny ", with " funny
 twice as big (in number of fortunes, not raw file size), saying
 .RS
 .sp
 .B fortune
-.I funny not-funny
+.I funny not\-funny
 .sp
 .RE
 will get you fortunes out of 
 .I funny
-two-thirds of the time.  The command
+two\-thirds of the time.  The command
 .RS
 .sp
 .B fortune
-.RI "90% " funny " 10% " not-funny
+.RI "90% " funny " 10% " not\-funny
 .sp
 .RE
 will pick out 90% of its fortunes from 
 .I funny
-(the ``10% not-funny'' is unnecessary, since 10% is all that's left).
+(the ``10% not\-funny'' is unnecessary, since 10% is all that's left).
 .PP
 The 
 .B \-e
@@ -200,27 +200,27 @@
 .RS
 .sp
 .B fortune \-e
-.I funny not-funny
+.I funny not\-funny
 .sp
 .RE
 is equivalent to
 .RS
 .sp
 .B fortune
-.RI "50% " funny " 50% " not-funny
+.RI "50% " funny " 50% " not\-funny
 .sp
 .RE
 This fortune also supports the BSD method of appending ``\-o'' to
 database names to specify offensive fortunes.  However this is
 .B not
-how fortune stores them: offensive fortunes are stored in a seperate
+how fortune stores them: offensive fortunes are stored in a separate
 directory without the ``\-o'' infix.  A plain name (i.e., not a path to a
 file or directory) that ends in ``\-o'' will be assumed to be an
 offensive database, and will have its suffix stripped off and be
 searched in the offensive directory (even if the neither of the
 .IR \-a " or " \-o
 options were specified).  This feature is not only for
-backwards-compatibility, but also to allow users to distinguish between
+backwards\-compatibility, but also to allow users to distinguish between
 inoffensive and offensive databases of the same name.
 .PP
 For example, assuming there is a database named
@@ -232,7 +232,7 @@
 .sp
 .B fortune
 90%
-.I definitions definitions-o
+.I definitions definitions\-o
 .RE
 .SH FILES
 Note: these are the defaults as defined at compile time.
Index: b/fortune/fortune-man.part2
===================================================================
--- a/fortune/fortune-man.part2	2012-07-07 02:27:47.000000000 +0200
+++ b/fortune/fortune-man.part2	2012-07-07 02:27:47.000000000 +0200
@@ -8,7 +8,7 @@
 .B fortune
 no longer finds the pointers file, it ignores the text file.
 .SH BUGS
-The division of fortunes into offensive and non-offensive by directory,
+The division of fortunes into offensive and non\-offensive by directory,
 rather than via the `\-o' file infix, is not 100% compatible with
 original BSD fortune. Although the `\-o' infix is recognised as referring
 to an offensive database, the offensive database files still need to be
@@ -38,7 +38,7 @@
 number of bug fixes and enhancements.
 .PP
 The original fortune/strfile format used a single file; strfile read the
-text file and converted it to null-delimited strings, which were stored
+text file and converted it to null\-delimited strings, which were stored
 after the table of pointers in the .dat file.  By NetBSD fortune 1.4,
 this had changed to two separate files: the .dat file was only the header
 (the table of pointers, plus flags; see
Index: b/fortune/fortune.6
===================================================================
--- a/fortune/fortune.6	2012-07-07 02:25:32.000000000 +0200
+++ b/fortune/fortune.6	2012-07-07 02:27:47.000000000 +0200
@@ -40,14 +40,14 @@
 .Dt FORTUNE 6
 .Os
 .Sh NAME
-.Nm fortune 
+.Nm fortune
 .Nd print a random, hopefully interesting, adage
 .Sh SYNOPSIS
 .Nm fortune
 .Op Fl aefilosw
-.Op Fl m Ar pattern 
-.Oo 
-.Op Ar N% 
+.Op Fl m Ar pattern
+.Oo
+.Op Ar N%
 .Ar file/dir/all
 .Oc
 .Sh DESCRIPTION
@@ -58,7 +58,7 @@
 is subdivided into those which are potentially offensive and those
 which are not.
 The options are as follows:
-.Bl -tag -width flag
+.Bl \-tag \-width flag
 .It Fl a
 Choose from all lists of maxims, both offensive and not.
 (See the
@@ -80,7 +80,7 @@
 for a description of patterns.
 .It Fl o
 Choose only from potentially offensive aphorisms.
-.Bf -symbolic
+.Bf \-symbolic
 Please, please, please request a potentially offensive fortune if and
 only if you believe, deep down in your heart, that you are willing
 to be offended.
@@ -89,14 +89,14 @@
 rather than give us
 grief about it, okay?)
 .Ef
-.Bd -filled -offset indent
+.Bd \-filled \-offset indent
 \&... let us keep in mind the basic governing philosophy
 of The Brotherhood, as handsomely summarized in these words:
-we believe in healthy, hearty laughter -- at the expense of
+we believe in healthy, hearty laughter \-\- at the expense of
 the whole human race, if needs be.
 Needs be.
-.Bd -filled -offset indent-two -compact
---H. Allen Smith, "Rude Jokes"
+.Bd \-filled \-offset indent\-two \-compact
+\-\-H. Allen Smith, "Rude Jokes"
 .Ed
 .Ed
 .It Fl s
@@ -133,40 +133,40 @@
 As an example, given two databases
 .Em funny
 and
-.Em not-funny ,
+.Em not\-funny ,
 with
 .Em funny
 twice as big, saying
-.Bd -literal -offset indent
-fortune funny not-funny
+.Bd \-literal \-offset indent
+fortune funny not\-funny
 .Ed
 .Pp
 will get you fortunes out of
 .Em funny
-two-thirds of the time.
+two\-thirds of the time.
 The command
-.Bd -literal -offset indent
-fortune 90% funny 10% not-funny
+.Bd \-literal \-offset indent
+fortune 90% funny 10% not\-funny
 .Ed
 .Pp
 will pick out 90% of its fortunes from
 .Em funny
-(the ``10% not-funny'' is unnecessary, since 10% is all that's left).
+(the ``10% not\-funny'' is unnecessary, since 10% is all that's left).
 The
 .Fl e
 option says to consider all files equal;
 thus
-.Bd -literal -offset indent
-fortune -e
+.Bd \-literal \-offset indent
+fortune \-e
 .Ed
 .Pp
 is equivalent to
-.Bd -literal -offset indent
+.Bd \-literal \-offset indent
 fortune 50% funny 50% not
-.Em -funny
+.Em \-funny
 .Ed
 .Sh FILES
-.Bl -tag -width Pa -compact
+.Bl \-tag \-width Pa \-compact
 .It Pa /usr/share/fortune
 .El
 .Sh SEE ALSO
