Description: Typo in a name.
Author: "Thomer M. Gil" <debian-bugs@thomer.com>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=361745

Index: b/datfiles/people
===================================================================
--- a/datfiles/people	2012-07-07 19:17:50.000000000 +0200
+++ b/datfiles/people	2012-07-07 19:22:01.000000000 +0200
@@ -714,7 +714,7 @@
 		-- Descartes, 1637
 %
 Conceit causes more conversation than wit.
-		-- LaRouchefoucauld
+		-- LaRochefoucauld
 %
 Confess your sins to the Lord and you will be forgiven;
 confess them to man and you will be laughed at.
@@ -1159,7 +1159,7 @@
 %
 Good advice is something a man gives when he is too old to set a bad
 example.
-		-- La Rouchefoucauld
+		-- La Rochefoucauld
 %
 Good judgement comes from experience.  Experience comes from bad judgement.
 		-- Jim Horning
@@ -3678,7 +3678,7 @@
 %
 We only acknowledge small faults in order to make it appear that we are
 free from great ones.
-		-- La Rouchefoucauld
+		-- La Rochefoucauld
 %
 We prefer to believe that the absence of inverted commas guarantees the
 originality of a thought, whereas it may be merely that the utterer has
