Description: backspaces characters were here because of a typo.
Author: Anthony DeRobertis <anthony@derobert.net>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=349867

Index: b/datfiles/computers
===================================================================
--- a/datfiles/computers	2012-07-07 19:17:50.000000000 +0200
+++ b/datfiles/computers	2012-07-07 19:19:21.000000000 +0200
@@ -1,6 +1,6 @@
 !07/11 PDP a ni deppart m'I  !pleH
 %
-101 USES FOR A DEAD MICROPROCESSOR
+101 USE SFOR A DEAD MICROPROCESSOR
 	(1)  Scarecrow for centipedes
 	(2)  Dead cat brush
 	(3)  Hair barrettes
@@ -3025,7 +3025,7 @@
 Our documentation manager was showing her 2 year old son around the office.
 He was introduced to me, at which time he pointed out that we were both
 holding bags of popcorn.  We were both holding bottles of juice.  But only
-*__he* had a lollipop.
+*he* had a lollipop.
 	He asked his mother, "Why doesn't HE have a lollipop?"
 	Her reply: "He can have a lollipop any time he wants to.  That's
 what it means to be a programmer."
