--- password-store-1.4.2/Makefile.orig	2013-07-03 22:39:47.000000000 -0400
+++ password-store-1.4.2/Makefile	2013-07-04 12:34:08.000000000 -0400
@@ -22,7 +22,8 @@
 #	@install -m 0644 -v contrib/pass.zsh-completion "$(DESTDIR)$(PREFIX)/share/zsh/site-functions/_pass"
 #
 #	Uncomment to install the fish completion file.
-#	@install -m 0644 -v contrib/pass.fish-completion "$(DESTDIR)$(PREFIX)/share/fish/completions/pass.fish"
+	mkdir -p "$(DESTDIR)$(PREFIX)/share/fish/completions"
+	@install -m 0644 -v contrib/pass.fish-completion "$(DESTDIR)$(PREFIX)/share/fish/completions/pass.fish"
 
 	@$(MAKE) -s install-platform
 
