--- password-store-1.4.2/Makefile.orig	2013-07-03 22:39:47.000000000 -0400
+++ password-store-1.4.2/Makefile	2013-07-04 12:31:31.000000000 -0400
@@ -19,7 +19,8 @@
 	@install -m 0644 -v contrib/pass.bash-completion "$(DESTDIR)$(SYSCONFDIR)/bash_completion.d/password-store"
 
 #	Uncomment to install the zsh completion file.
-#	@install -m 0644 -v contrib/pass.zsh-completion "$(DESTDIR)$(PREFIX)/share/zsh/site-functions/_pass"
+	@mkdir -p "$(DESTDIR)$(PREFIX)/share/zsh/site-functions"
+	@install -m 0644 -v contrib/pass.zsh-completion "$(DESTDIR)$(PREFIX)/share/zsh/site-functions/_pass"
 #
 #	Uncomment to install the fish completion file.
 #	@install -m 0644 -v contrib/pass.fish-completion "$(DESTDIR)$(PREFIX)/share/fish/completions/pass.fish"
